File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = []
    for i in s:
        if i == '(':
            answer.append(i)
        else:
            if answer:
                answer.pop()
            else:
                return False
        
    if len(answer) == 0 :
        return True
    else:
        return False
    

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var stack = [];
    if (s[0]==')'){
        return false;
    }
    for (let i=0; i<s.length; i++){
        if (s[i] == '('){
            stack.push(s[i])    
        }
        if (s[i] == ')'){
            stack.pop()
        }
    }
    if (stack.length ==0){
        return true;
    }
    if (stack.length !=0){
        return false;
    }
    
}